### PR TITLE
Printing for MPolyAnyMap

### LINKED ISSUE #2773: Adjust printing of `MPolyAnyMap` to follow OSCAR printing guidelines
... as described in https://docs.oscar-system.org/dev/DeveloperDocumentation/printing_details/ (with fixes in PR #2769).

Right now I see this:
```
julia> psi = hom(R, R, [x^2, y^3])
Map with following data
Domain:
=======
Multivariate polynomial ring in 2 variables over QQ
Codomain:
=========
Multivariate polynomial ring in 2 variables over QQ

julia> print(psi)
Map with following data
Domain:
=======
Multivariate polynomial ring in 2 variables over QQ
Codomain:
=========
Multivariate polynomial ring in 2 variables over QQ
julia> print(IOContext(stdout, :supercompact => true), psi)
Map with following data
Domain:
=======
Multivariate polynomial ring
Codomain:
=========
Multivariate polynomial ring
```

Based on [this discussion](https://hackmd.io/cwOoBrTaReihNtmFjSdtTg) I think it rather should look something like this (up to details, of course, feel free to modify):
```
julia> psi  # detailed
Ring homomorphism 
  from Multivariate polynomial ring in 2 variables over QQ
  to Multivariate polynomial ring in 2 variables over QQ
defined by 
 x -> x^2
 y -> y^3

julia> print(psi)  # oneline
Hom: Multivariate polynomial ring -> Multivariate polynomial ring

julia> print(IOContext(stdout, :supercompact => true), psi)
Ring homomorphism
```

### PR DESCRIPTION
**Pull request by simonbrandhorst**
Resolves #2773

### CODE REVIEW

File: src/Rings/MPolyMap/MPolyRing.jl [hunk 1]
```diff
@@ -69,13 +70,20 @@ julia> Qi, i = quadratic_field(-1)
...
+  x -> x^2
+  y -> y^2
+with map on coefficients
+  Map with following data
+    Domain:
+    =======
+    Qi
+  Codomain:
+    =========
+    Qi
```
Comment (lgoettgens) [inline src/Rings/MPolyMap/MPolyRing.jl:86]: There is something weird going on with indentation here.

Comment (simonbrandhorst) [reply src/Rings/MPolyMap/MPolyRing.jl:86]: It is exactly as intended. The problem is just that the map on coefficients does not follow the new printing guidelines.  There is a PR in AbstractAlgebra addressing this.

Comment (lgoettgens) [reply src/Rings/MPolyMap/MPolyRing.jl:86]: I would have expected `Domain` and `Codomain` to be in the same level

Comment (simonbrandhorst) [reply src/Rings/MPolyMap/MPolyRing.jl:86]: You are right, sorry.

**Comment by simonbrandhorst** [comment]
This should probably be based on #2796

**Comment by lgoettgens** [comment]
If I understood @simonbrandhorst correctly, he wanted to wait until #2796 is merged and rebase then.

**Comment by thofma** [comment]
He wrote me that he wants it to be merged now.

**Comment by simonbrandhorst** [comment]
Thought it to be easier like that in the end :-).

**Comment by lgoettgens** [comment]
Ok. Then I'll try to rebase #2796 on top of this.

**Pull request merged**